bloomfilter/bloom_filter.go

```
@@ -0,0 +1,51 @@
+package bloomfilter
+
+import (
+	"errors"
+	"fmt"
+	"github.com/spaolacci/murmur3"
+	"prac/bitmap"
+)
+
+type BloomFilter struct {
+	size int
+	hash int
+	b    bitmap.BitMap
+}
+
+func NewBloomFilter(siz int) (BloomFilter, error) {
+
+	bmp, err := bitmap.NewBitmap(siz)
+	if err != nil {
+		fmt.Println("The bloom filter size is wrong")
+		return BloomFilter{0, 3, bmp}, errors.New("wrong size error")
+	} else {
+		return BloomFilter{siz, 3, bmp}, nil
+	}
+}
+
+func (bf BloomFilter) Put(st string) error {
+	value1 := (int(murmur3.Sum32WithSeed([]byte(st), 123))) % bf.size
+	value2 := (int(murmur3.Sum32WithSeed([]byte(st), 124))) % bf.size
+	value3 := (int(murmur3.Sum32WithSeed([]byte(st), 125))) % bf.size
+	bf.b.Set(value1)
+	bf.b.Set(value2)
+	bf.b.Set(value3)
+
+	return nil
+}
+func (bf BloomFilter) Mightcontain(st string) (bool, error) {
+	value1 := (int(murmur3.Sum32WithSeed([]byte(st), 123))) % bf.size
+	value2 := (int(murmur3.Sum32WithSeed([]byte(st), 124))) % bf.size
+	value3 := (int(murmur3.Sum32WithSeed([]byte(st), 125))) % bf.size
+
+	flag1, _ := bf.b.Get(value1)
```

Don't ignore the error. This will give you an incorrect answer